src/main/java/uk/gov/hmcts/darts/cases/service/impl/CaseServiceImpl.java

```diff
@@ -102,11 +102,15 @@ public List<Hearing> getCaseHearings(Integer caseId) {
 
         List<HearingEntity> hearingList = hearingRepository.findByCaseIds(List.of(caseId));
 
-        if (hearingList.isEmpty()) {
-            getCourtCaseById(caseId);
+        if (hearingList.isEmpty() && !caseRepository.existsById(caseId)) {
```

Comment:
Do we want this caseRepository check?
If no hearing exist, then that means that no events appeared, which means that the case has been added by an addCase request, but never happened. Do we want to return an empty array in this scenario?

Reply:
As discussed, this is existing functionality implemented in DMP-753 that we don't want to change as part of this PR. That ticket specifically says to change this from an empty response to an error.